### ISSUE TITLE
nix-hash base32 conversion port to javascript

### ISSUE THREAD

**Issue by jbboehr** [svanderburg/node2nix#93]
I've been working on an alternative implementation of node2nix and I noticed the majority of the execution time was being spent running `nix-hash` as a process, so I've ported the base32 encoding algorithm to Javascript (TypeScript, actually). Feel free to use it if you'd like, it may speed up execution time.

The original code is here: https://github.com/NixOS/nix/blob/27d1c052ae4e53328c2909b040e204bb7f57ff96/src/libutil/hash.cc#L76

And my port is here: https://gist.github.com/jbboehr/6fa20f57a031e8dd2c2b23234e6582e4

**Comment by svanderburg** [comment]
Nice effort. Aside from performance, it would be helpful to get rid on the `nix-hash` dependency for usability reasons. I know quite a few people who try to run `node2nix` on a machine without Nix installed, and they run into problems all the time.

I'll have a look at it soon.

**Comment by svanderburg** [comment]
Hi,

I just "solved" the issue by using a different strategy. A while ago, I created an issue in the Nix repo to support hashes in base64 notation (https://github.com/NixOS/nix/commit/c0015e87af70f539f24d2aa2bc224a9d8b84276b). The good news is that this functionality is supported by Nix 2.0. The only thing is: I filed this issue a long time ago and forgot about it, since it still took a while before Nix 2.0 was released.

This means that this silly conversion stuff is no longer required. We can simply take the SHA512 hashes in base64 notation from NPM. Nix 2.0 and higher can also use these values without any conversions. In revision a9b41be9482ddbc62e9330d2039da98cc59eab70, I removed the `nix-hash` invocation that does the conversion.

In addition to the fact that this commit removes code that is no longer required, we also don't have to solve any licensing problems (MIT vs LGPLv2.1+).
